Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -154,8 +187,45 @@ AppendPipeline::AppendPipeline(SourceBufferPrivateGStreamer& sourceBufferPrivate
         m_demux = makeGStreamerElement("matroskademux"_s);
         m_typefind = makeGStreamerElement("identity"_s);
     } else if (type == "audio/mpeg"_s) {
-        m_demux = makeGStreamerElement("identity"_s);
+        // Will be instantiated later based on typefind results.
+        m_demux = nullptr;
         m_typefind = makeGStreamerElement("typefind"_s);
+
+        g_signal_connect(m_typefind.get(), "have-type", G_CALLBACK(+[](
+            GstElement* typefind, guint, GstCaps* caps, AppendPipeline* appendPipeline) {
+            ASSERT(!isMainThread());
+
+            // We don't want to create the demuxer twice if the type changes for whatever reason,
+            // and this is cheaper than storing the handler id and disconnecting the handler.
+            if (appendPipeline->m_demux)
+                return;
+
+            auto capsString = String::fromUTF8(GUniquePtr<char>(gst_caps_to_string(caps)).get());
+            GST_DEBUG_OBJECT(appendPipeline->pipeline(), "Creating demuxer for %s caps", capsString.utf8().data());
+            if (capsString.startsWith("application/x-id3"_s))
+                appendPipeline->m_demux = makeGStreamerElement("id3demux"_s);
+            else if (capsString.startsWith("audio/mpeg"_s))
+                appendPipeline->m_demux = makeGStreamerElement("identity"_s);
```

Comment:
There is no need for a conversion to string. Get the first structure from the caps and use `gst_structure_has_name()`

Reply:
Done, but still, when I later need to print the name by using `gst_structure_get_name()`, webkit-style-check nags me about using `StringView gstStructureGetName()`. This forces me to use `gstStructureGetName(capsStructure).toStringWithoutCopying().utf8().data()`. I don't know how wasteful all those conversions are. :-/